src/object/cli_shard.c

```diff
@@ -459,14 +459,10 @@ dc_rw_cb(tse_task_t *task, void *arg)
 
 		/* update the sizes in iods */
 		for (i = 0; i < orw->orw_nr; i++) {
-			if (!is_ec_obj || reasb_req->orr_fail == NULL ||
-			    iods[i].iod_size == 0)
-				iods[i].iod_size = sizes[i];
-			if (is_ec_obj && reasb_req->orr_recov &&
-			    reasb_req->orr_fail->efi_uiods[i].iod_size == 0) {
+			iods[i].iod_size = sizes[i];
+			if (is_ec_obj && reasb_req->orr_recov)
 				reasb_req->orr_fail->efi_uiods[i].iod_size =
 					sizes[i];
```

Comment:
this change looks incorrect. When the server has nothing for this akey, it will set the size as zero, but for EC we cannot directly reset user iod_size as zero for this case.

Reply:
in EC degraded mode. (orr_fail != NULL), and if iod_size != 0, then it will not be updated.

Reply:
Hmm, we probably should set sizes to be 0 in this case. and do not update iod_size if size is 0. I will update the patch.